lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -16,6 +16,9 @@
  * 4. Return all those items in one handy bundle.
  */
 
+/** @typedef {Omit<LH.Artifacts.TraceTimes, 'firstContentfulPaint'> & {firstContentfulPaint?: number}} TraceTimesWithoutFCP */
+/** @typedef {Omit<LH.Artifacts.TraceOfTab, 'firstContentfulPaintEvt'|'timings'|'timestamps'> & {timings: TraceTimesWithoutFCP, timestamps: TraceTimesWithoutFCP, firstContentfulPaintEvt?: LH.Artifacts.TraceOfTab['firstContentfulPaintEvt']}} TraceOfTabWithoutFCP */
```

Comment:
these types seem kind of terrible for downstream users, but I guess we can iterate on it in the future :)

Reply:
heh, yeah there will be lots of interesting problems to solve when we expose these types :)